From ba25f40fdafa61c16ff9914da8abed85ae027f26 Mon Sep 17 00:00:00 2001
From: Jake Day <jake@ninebysix.com>
Date: Fri, 20 Oct 2017 13:50:48 -0400
Subject: [PATCH] surface laptop hid id is 0904 for hid input (f001 is the
 parent)

---
 drivers/hid/hid-ids.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/hid-ids.h b/drivers/hid/hid-ids.h
index e011419cc..b992a50b2 100644
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -769,7 +769,7 @@
 #define USB_DEVICE_ID_MS_TYPE_COVER_PRO_4	0x07e8
 #define USB_DEVICE_ID_MS_TYPE_COVER_PRO_4_1	0x07e4
 #define USB_DEVICE_ID_MS_SURFACE_BOOK		0x07cd
-#define USB_DEVICE_ID_MS_SURFACE_LAPTOP		0xf001
+#define USB_DEVICE_ID_MS_SURFACE_LAPTOP		0x0904
 #define USB_DEVICE_ID_MS_POWER_COVER		0x07da
 
 #define USB_VENDOR_ID_MOJO		0x8282
