From 5cb50cf9b9bd1dd50df3634808d040cd5465f46c Mon Sep 17 00:00:00 2001
From: Jake Day <jake@ninebysix.com>
Date: Wed, 8 Nov 2017 21:58:30 -0500
Subject: [PATCH] updating multitouch driver to specify touchscreen devices

---
 drivers/hid/hid-multitouch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hid/hid-multitouch.c b/drivers/hid/hid-multitouch.c
index 1cfbd26fb..ba5569dbf 100644
--- a/drivers/hid/hid-multitouch.c
+++ b/drivers/hid/hid-multitouch.c
@@ -1172,7 +1172,7 @@ static int mt_input_configured(struct hid_device *hdev, struct hid_input *hi)
 			__set_bit(BTN_STYLUS, hi->input->keybit);
 			break;
 		case HID_DG_TOUCHSCREEN:
-			/* we do not set suffix = "Touchscreen" */
+			suffix = "Touchscreen";
 			break;
 		case HID_DG_TOUCHPAD:
 			suffix = "Touchpad";
